include/zephyr/fs/fs.h

```diff
@@ -672,6 +672,21 @@ int fs_register(int type, const struct fs_file_system_t *fs);
  */
 int fs_unregister(int type, const struct fs_file_system_t *fs);
 
+/**
+ * @brief Attempt to proactively clean file system
+ *
+ * Returns a negative error code on failure.
+ * Ignored cleaning request is not a failure.
+ *
+ * @param path Path to the mounted directory
+ *
+ * @retval 0 on success;
+ * @retval -EINVAL when a bad path to a directory, or a file, is given;
+ * @retval -ENOTSUP when not implemented by underlying file system driver;
+ * @retval <0 an other negative errno code on error.
+ */
+int fs_gc(const char *path);
```

Comment:
Rename this to something like 'fs_gc_by_path' or similar; I would rather have short name, default one, to be reserved for gc by mount point object pointer.

Reply:
Changed logic, so now `fs_gc` accepts mount point pointer.